torch/_inductor/fx_passes/post_grad.py

```diff
@@ -627,44 +630,69 @@ def reinplace_scatters(graph):
     for node in reversed(graph.nodes):
         storage_to_nodes[get_node_storage(node)].append(node)
         if node.target == aten.copy_.default:
-            copy_args_to_copy_nodes[(node.args[0], node.args[1])] = node
+            src = node.args[0]
+            dst = node.args[1]
+            # If the target is a getitem and it indexes a possible clone,
+            # then skip over it
+            if (
+                dst.target == operator.getitem
+                and dst.args[0].kwargs["kwargs"][dst.args[1]] == node.args[0]
+            ):
+                dst = dst.args[0]
+            copy_args_to_copy_nodes[(src, dst)] = node
             assert node.args[0].op == "placeholder"
             mutated_inputs.add(node.args[0])
 
+    def can_replace(node, mutated_arg):
+        if get_node_storage(mutated_arg) is None:
+            return False
+        shared_view_nodes = storage_to_nodes[get_node_storage(mutated_arg)]
+        if mutated_arg.op == "placeholder":
+            if not (
+                copy_node := copy_args_to_copy_nodes.get((mutated_arg, node), False)
+            ):
+                return False
+
+            if len(shared_view_nodes) > 2:  # Arg aliases another node other than copy_
```

Comment:
This check is probably also too liberal. if there happen to be any views of the input in the graph, then this will also be > 2 and we won't reinplace.

Handling the aliasing case is a bit tricky though (but definitely feels solvable). Do we want to try to handle reinplacing in 100% of cases as part of this PR?

Reply:
In particular: we'd ideally like to remove the `copy_()` on `x` for a case like this:
```
def f(x, out):
    x_view = x.view(-1)
    triton_kernel[grid](inp=x, out=out)
    out2 = x_view.mul(2)
    return out, out2
```

But we want to make sure **not** to remove the `copy_()` for a case like this (where we actually do mutate the alias of x later):
```
def f(x, out):
    x_view = x.view(-1)
    triton_kernel[grid](inp=x, out=out)
    x_view.mul_(2)
    return out
```

Reply:
yup, this could be a good follow up too. For now, i kept the reinplace_scatters as is